document-store/src/main/java/org/hypertrace/core/documentstore/Collection.java

```diff
@@ -102,6 +104,14 @@
    */
   CreateResult create(Key key, Document document) throws IOException;
 
+  /**
+   * Updates existing documents if the corresponding Filter condition evaluates to true
+   * @param documents to be updated in bulk
+   * @return true if operation succeeded
+   */
+  boolean bulkUpdate(List<Triple<Key, Document, Filter>> documents);
```

Comment:
Can we return `BulkUpdateResult` the way we did in the newer API, so it will be easy to extend?

Reply:
fixed